ondewo/nlu/intent.proto

```diff
@@ -940,38 +943,54 @@ message IntentSorting {
     SortingMode sorting_mode = 2;
 }
 
-message CreateTrainingPhraseRequest {
-    // Required. The name of the intent for which to create a training phrase for.
-    // Needed since the name of the training phrase may not be set upon creation.
-    // Format: `projects/<Project ID>/agent/intents/<Intent ID>`.
-    string intent_name = 1;
+message UpdateTrainingPhraseBatchRequest {
+    repeated Intent.TrainingPhrase training_phrases = 1;
+}
 
-    // Required. The training phrase to create.
-    Intent.TrainingPhrase training_phrase = 2;
+message TrainingPhraseStatus {
+    oneof phrase_or_status {
+        Intent.TrainingPhrase training_phrase = 1;
+        string error_message = 2;
+    }
+}
 
-    // Required. The language of training phrases
-    string language_code = 3;
+message TrainingPhraseStatusBatchResponse {
+    repeated TrainingPhraseStatus training_phrase_statuses = 1;
+    // indicates if statuses of some of the training phrases have errors
+    bool has_errors = 2;
 }
 
-message GetTrainingPhraseRequest {
-    // Required. The name of the training phrase.
-    // Format: `projects/<Project ID>/agent/intents/<Intent ID>/trainingPhrases/<Training Phrase ID>`
-    string name = 1;
+message CreateTrainingPhraseBatchRequest {
 
-    // Required. The language of training phrases
-    string language_code = 2;
-}
+    message CreateTrainingPhraseRequest {
+        // Required. The name of the intent for which to create a training phrase for.
+        string intent_name = 1;
 
-message UpdateTrainingPhraseRequest {
-    // Required. The training phrase to update.
-    Intent.TrainingPhrase training_phrase = 1;
+        // Required. The training phrase to create.
+        Intent.TrainingPhrase training_phrase = 2;
+    };
+    repeated CreateTrainingPhraseRequest training_phrase_requests = 1;
+}
 
-    // Required. The language of training phrases
-    string language_code = 2;
+message GetTrainingPhraseBatchRequest {
+    // Required. The names of the training phrases.
+    // Format: `projects/<Project ID>/agent/intents/<Intent ID>/trainingPhrases/<Training Phrase ID>`
+    repeated string names = 1;
 }
 
-message DeleteTrainingPhraseRequest {
-    // Required. The name of the training phrase.
+message DeleteTrainingPhraseBatchRequest {
+    // Required. The names of the training phrases.
     // Format: `projects/<Project ID>/agent/intents/<Intent ID>/trainingPhrases/<Training Phrase ID>`
-    string name = 1;
-}
\ No newline at end of file
+    repeated string names = 1;
+}
+
+message DeleteTrainingPhraseBatchResponse {
+    message DeleteTrainingPhraseStatus{
+        oneof delete_status{
+            google.protobuf.Empty successfully_deleted = 1;
```

Comment:
empty? 
how does that work?

does it get defined automatically if the error is not set?